pyccel/ast/numpyext.py

```diff
@@ -2087,6 +2090,77 @@ def __new__(cls, a, axis = None):
 
         return PyccelArrayShapeElement(a, axis)
 
+class NumpyIsNan(PyccelInternalFunction):
+    """ 
+    Represents a call to numpy.isnan() function.
+
+    This class encapsulates a call to the Numpy 'isnan' function. It is used to
+    check whether the elements of a given array or expression are NaN (Not-a-Number).
+
+    Parameters
+    ----------
+    arg: A Pyccel expression or array to be checked for NaN values.
+
+    See Also
+    --------
+    numpy.isnan :
+        See NumPy docs : <https://numpy.org/doc/stable/reference/generated/numpy.isnan.html>.
+    """
```

Comment:
Thanks for fixing the docstrings. We use numpy docstring style (https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes) which also requires the types of the arguments:

```suggestion
    """ 
    Represents a call to numpy.isnan() function.

    This class encapsulates a call to the Numpy 'isnan' function. It is used to
    check whether the elements of a given array or expression are NaN (Not-a-Number).

    Parameters
    ----------
    arg : PyccelAstNode
        A Pyccel expression or array to be checked for NaN values.

    See Also
    --------
    numpy.isnan :
        See NumPy docs : <https://numpy.org/doc/stable/reference/generated/numpy.isnan.html>.
    """
```